False positive - cgo - block starting/ending with whitespaces

For the newest wsl (v4.5.0), given this code snippet: 
```go
// SetEventAlarm updates the EventAlarm settings.
func (e *Ecopoint) SetEventAlarm(alarm EventAlarm) error {
	ret := C.ecopointSetEventAlarm(
		e.instance,
		C.uint8_t(alarm.Count), C.uint8_t(alarm.BaseTime), C.uint8_t(alarm.LockTime), C.uint16_t(alarm.Threshold),
	)
	if ret != 0 {
		return getLastError()
	}

	return nil
}
```

I'm getting the following false positives:
```
ecopoint_clib.go:117:27: block should not start with a whitespace
ecopoint_clib.go:120:68: block should not end with a whitespace (or comment)
```

Pointing to the start and end of the `ret` block. 

I'm still trying to isolate the issue, but I've not been able to produce it yet outside of our project.

If I look at the examples in #166, the issue might in fact be related to `unsafe.Pointer`. I can do a repro with this:

```go
package testpkg                              
                                             
/*                                           
#include <stdlib.h>                          
*/                                           
import "C" regenerate cgo definitions        
                                             
import (                                     
    "fmt"                                    
    "unsafe"                                 
)                                            
                                             
func _() {                                   
    cs := C.CString("Hello CString\n")       
    defer C.free(unsafe.Pointer(cs))         
                                             
    fmt.Println("Hello, world")              
}                                            
```

```sh
› wsl ./...
/home/simon/.../cgo.go:16:2: defer statements should only be cuddled with expressions on same variable
```

I was under the impression that what's being analyzed here is the actual Go file, but I'm not 100% sure anymore. Maybe it is in fact the built Go file with `line` directives. If I only pick files with non adjusted position, I don't get any false positives, but I also don't get any diagnostics for the original Go file.

If we are in fact looking at the generated file, this is what the generated CGO code look like:

```sh
› go tool cgo cgo.go
› cat _obj/cgo.cgo1.go
// Code generated by cmd/cgo; DO NOT EDIT.

//line /home/simon/.../cgo/cgo.go:1:1
package testpkg

/*
#include <stdlib.h>
#include "cgo.h"
*/
import _ "unsafe"

import (
        "fmt"
        "unsafe"
)

func _() {
        cs := ( /*line :15:8*/_Cfunc_CString /*line :15:16*/)("Hello CString\n")
        defer func() func() { _cgo0 := /*line :16:15*/unsafe.Pointer(cs); return func() { _cgoCheckPointer(_cgo0, nil); /*line :16:34*/_Cfunc_free(_cgo0); }}()()

        fmt.Println("Hello, world")
}
```

Or to be more readable:

```go
cs := (_Cfunc_CString)("Hello CString\n")
defer func() func() {
    _cgo0 := unsafe.Pointer(cs)
    return func() {
        _cgoCheckPointer(_cgo0, nil)
        _Cfunc_free(_cgo0)
    }
}()()
```

I don't think we should ever analyze generated Go code with this analyzer because it makes no sense to look at style for generated code. Especially not with file and line directives.

--- a/analyzer.go
+++ b/analyzer.go
@@ -86,8 +86,12 @@ func (wa *wslAnalyzer) run(pass *analysis.Pass) (interface{}, error) {
 			continue
 		}
 
-		// if the file is related to cgo the filename of the unadjusted position is a not a '.go' file.
 		fn := pass.Fset.PositionFor(file.Pos(), false).Filename
+
+		// if the file is related to cgo the filename of the unadjusted position is a not a '.go' file.
+		if !strings.HasSuffix(fn, ".go") {
+			continue
+		}
 
 		// The file is skipped if the "unadjusted" file is a Go file, and it's a generated file (ex: "_test.go" file).
 		// The other non-Go files are skipped by the first 'if' with the adjusted position.

--- a/testdata/src/default_config/cgo/cgo.go
+++ b/testdata/src/default_config/cgo/cgo.go
@@ -22,16 +22,16 @@ func _() {
 
 func Advanced() {
 	var foo = 1
-	var bar = 2 // want "declarations should never be cuddled"
-	var biz int // want "declarations should never be cuddled"
+	var bar = 2
+	var biz int
 
 	x := []string{}
 	x = append(x, "literal")
 	notUsed := "just assigning, don't mind me"
-	x = append(x, "not z..") // want "append only allowed to cuddle with appended value"
+	x = append(x, "not z..")
 	useMe := "right away"
 	alsoNotUsed := ":("
-	x = append(x, "just noise") // want "append only allowed to cuddle with appended value"
+	x = append(x, "just noise")
 	x = append(x, useMe)
 
 	_ = foo

